### PR TITLE
Fix deserialization segfault in bionic.

### PR DESCRIPTION
**Pull request by mjcarroll**
This should fix the segfaults that we were seeing on Bionic with rclcpp.

Connects to ros2/rclcpp#464

### CODE REVIEW

**Comment by mjcarroll** [comment]
* Linux [Build Status](http://ci.ros2.org/job/ci_linux/4320/)
* Linux-aarch64 [Build Status](http://ci.ros2.org/job/ci_linux-aarch64/1312/)
* macOS [Build Status](http://ci.ros2.org/job/ci_osx/3549/)
* Windows [Build Status](http://ci.ros2.org/job/ci_windows/4393/)

**Comment by mjcarroll** [comment]
[Build Status](https://ci.ros2.org/job/ci_linux/4324/)

**Comment by mjcarroll** [comment]
* Linux (xenial)  [Build Status](http://ci.ros2.org/job/ci_linux/4331/)
* Linux-aarch64 [Build Status](http://ci.ros2.org/job/ci_linux-aarch64/1319/)
* macOS [Build Status](http://ci.ros2.org/job/ci_osx/3556/)
* Windows [Build Status](http://ci.ros2.org/job/ci_windows/4404/)

**Comment by mjcarroll** [comment]
I have opened a new issue to track the test hanging, as it happens without this patch as well.

ros2/rclcpp#469

File: rmw_fastrtps_cpp/include/rmw_fastrtps_cpp/TypeSupport_impl.hpp [hunk 1]
```diff
@@ -420,6 +420,32 @@ void deserialize_field(
...
+      // passing it as a reference to Fast-CDR.
+      new (field) std::string();
+    }
+    deser >> *static_cast<std::string *>(field);
+  } else if (member->array_size_ && !member->is_upper_bound_) {
+    deser.deserializeArray(static_cast<std::string *>(field), member->array_size_);
+  } else {
+    auto & vector = *reinterpret_cast<std::vector<std::string> *>(field);
+    if (call_new) {
+      new(&vector) std::vector<std::string>;
```
Comment (mikaelarguedas) [inline rmw_fastrtps_cpp/include/rmw_fastrtps_cpp/TypeSupport_impl.hpp]: Nitpick: I don't think we have a guideline for this but above you have a whitespace before the parenthesis and here there is not. We should pick one and us it in both places.

Comment (mjcarroll) [reply rmw_fastrtps_cpp/include/rmw_fastrtps_cpp/TypeSupport_impl.hpp]: It appears that everywhere else in the file doesn't have the space, so I'll follow that convention here.  Interesting that cpplint or uncrustify didn't catch it.

**Pull request merged**